stack.c

```diff
@@ -3,57 +3,60 @@
 
 #include "stackconfig.h"
 #include "stack.h"
+void init(Stack* stack) {
+    stack->top = -1;
+}
+void push(Stack* stack, stackType input) {
+
+    if (isFull(stack)) {
+        {
+            printf("Stack overflow while push()");
+            return;
+        }
+        stack->top = stack->top + 1;
+        stack->data[stack->top] = input;
 
-void push(Stack* stack, stackType data) {
-    Node* newNode = malloc(sizeof(Node));
-    if(newNode == NULL) {
-        printf("Stack overflow while push()");
-        return;
+        //newNode->data = data;
+        //newNode->next = stack->top;
+        //stack->top = newNode;
     }
-    newNode->data = data;
-    newNode->next = stack->top;
-    stack->top = newNode;
 }
+    stackType pop(Stack * stack) {
+        if (isEmpty(stack)) {
+            printf("Stack underflow while pop\n");
+            return -1;
+        }
+        stackType a = stack->data[stack->top];
+        stack->top = stack->top - 1;
+        return a;
+    }
 
-stackType pop(Stack* stack) {
-    if(stack->top == NULL) {
-        printf("Stack underflow while pop\n");
-        return -1;
+    stackType peek(Stack * stack) {
+        if (isEmpty(stack)) {
+            printf("Stack underflow while pop\n");
+            return -1;
+        }
+        stackType a = stack->data[stack->top];
+        return a;
     }
-    Node* next = stack->top->next;
-    stackType data = stack->top->data;
-    free(stack->top);
-    stack->top = next;
-    return data;
-}
 
-stackType peek(Stack* stack) {
-    if(stack->top == NULL) {
-        printf("Stack underflow while pop\n");
-        return -1;
+    int isEmpty(Stack * stack) {
+        return stack->top == -1;
     }
-    stackType data = stack->top->data;
-    return data;
-}
 
-int isEmpty(Stack* stack) {
-    return stack->top == NULL;
-}
+    int isFull(Stack * stack) {
 
-int isFull(Stack* stack) {
-    Node* newNode = malloc(sizeof(Node));
-    if(newNode == NULL) {
-        return 1;
+        if (stack->top == 99) {
```

Comment:
Avoid magic numbers like 99 in your code. Use a macro or a variable

Reply:
it can be replaced by MAX-1